lib/log.dart

```diff
@@ -1,3 +1,6 @@
+import 'package:flutter/material.dart';
+
+import 'widgets/app.dart';
```

Comment:
This `lib/log.dart` file is one that virtually the whole app imports, including parts of `lib/api/`. So I'd like to keep it from depending on the app's widgets.

That's the motivation for the indirection found in my draft branch that IIRC you started from:
https://github.com/zulip/zulip-flutter/compare/main...gnprice:zulip-flutter:dev-retry#diff-055b57342eabea1b640e04ca24eb50bc75d0c61383b928ad7211e37159a6c57bR34-R42
https://github.com/zulip/zulip-flutter/commit/68682403587804ab6dd392491fd7700596972ea3#diff-4c278608930a95a1cba05bce5a0f7d8f43bae364ed1e722eca5d8c1c4b29b57aR94-R101

where this file just has a mutable callback that defaults to something simple, and then ZulipApp can set the callback to something more complex after it starts up.

(In fact this file doesn't end up using these imports, right? It still has that indirection. I suspect the reason the linter hasn't complained about them is just the references in dartdoc.)

Reply:
> In fact this file doesn't end up using these imports, right?

Right, these imports were included only for the dart doc references. Removing them and updating the commit message to reflect the motivation:

```diff
-    The imports in `lib/log.dart` are for the code references in
-    the reportErrorToUser* helpers' dart doc.
-
-    The motivation of having these set of helpers, rather than using
-    showErrorDialog and showSnackBar directly, is that we make displaying
-    errors more accessible in non-widget code.
+    The motivation of having this indirection, rather than using
+    `showErrorDialog` and `showSnackBar` directly, is to keep dependencies
+    of `lib/log.dart` from relying on widget code.    
```